7.0 fix timesheet fulfill crash

gurneyalex opened the pull request:
fix a crash with the timesheet fulfill wizard which was passing datetime objects instead of strings 

add a test showing the bug is fixed.

gurneyalex committed:
add missing dependency on `project`

gurneyalex committed:
fix crash when using timesheet_fulfill wizard

The wizard would crash because the call to create new attendance
was passed a datetime object, not a string.

Fix the code + add test showing the bug is fixed.

pedrobaeza commented on hr_timesheet_fulfill/wizard/timesheet_fulfill.py:
```diff
@@ -134,8 +133,7 @@ def fulfill_timesheet(self, cr, uid, ids, context=None):
                 att_end = {
-                    'name': datetime.strptime(datetime_current + date_end,
-                                              '%Y-%m-%d %H:%M:%S'),
+                    'name': datetime_current + date_end,
```
There is one missing space between both strings.

gurneyalex replied:
@pedrobaeza I must be tired, I really don't see what you mean there. Could you clarify?

pedrobaeza replied:
OK, don't worry, I try to explain:

datetime_current = "2014-01-01"
date_end = "22:46:00"

datetime_current + date_end = "2014-01-0122:46:00", not "2014-01-01 22:46:00"

Regards.

gurneyalex replied:
I'm quite certain there is a space after the opening quote on line 134

pedrobaeza replied:
Sorry, I didn't see it. I only see the concatenation of the two strings.

lepistone replied:
:weary: I have a bit of a headache, you're saying `date_end` is actually a time, and `datetime_current` is actually a date? (string)

gurneyalex committed:
Update __openerp__.py

fix indentation and spaces

gurneyalex commented:
It looks like the added test is triggering an error in `hr_attendance_analysis`. I'll look into this and try to provide a fix.

gurneyalex commented:
the error was there before: it is a bug in the test which only fires if the tests are run after the 11th of the month... See #10 for the fix.

eLBati commented on hr_timesheet_fulfill/test/timesheet_fulfill.yml:
```diff
@@ -0,0 +1,29 @@
+  !python {model: hr.timesheet.fulfill }: |
+    context['active_id'] = ref('test_timesheet')
+    ret = self.fulfill_timesheet(cr, uid, [ref('wizard_fulfill')], context=context)
+    assert ret == {'type': 'ir.actions.act_window_close'}
```
Is it possible to make the test check the created data?

gurneyalex replied:
@elBati: you're right, I was focusing on the crash when writing the test. I'll update this.

gurneyalex committed:
fix hr.attendance creation

in v7, the field sheet_id of hr.attendance is a function field, so we cannot
write to it.
However, passing a sheet_id key in the context at create time helps the search
a bit.

gurneyalex committed:
test timesheet is created with correct values

this required changing the employee, because of
https://github.com/odoo/odoo/issues/2011 which makes employee_fp unsuitable for
testing, and fixing the user used to run the wizard in the test (because the
uid is used to find the timesheet_sheet in the hr.attendance sheet_id function
field).

gurneyalex committed:
pep8 fix

gurneyalex commented:
Travis is read because of 80char lines in other modules. See #12 for this.

eLBati commented on hr_timesheet_fulfill/wizard/timesheet_fulfill.py:
```diff
@@ -123,23 +127,23 @@ def fulfill_timesheet(self, cr, uid, ids, context=None):
-                attendance_obj.create(cr, uid, att_start, context)
-                attendance_obj.create(cr, uid, att_end, context)
+                attendance_obj.create(cr, uid, att_start,
+                                      ontext=attendance_context)
```
TypeError: create() got an unexpected keyword argument 'ontext'

gurneyalex committed:
fix typo

eLBati commented:
@gurneyalex please rebase. Thanks

gurneyalex commented:
Rebased. Travis is red because of the new pylint config (handled in #16).

guewen commented:
:+1:

mdietrichc2c commented:
:+1:

lepistone commented:
I merged #16 and restarted travis.

lepistone commented:
Ok, you didn't introduce the weird variable names, you're fixing other stuff. :+1: , and I'm merging, thanks!
